assets/src/hooks/useDetour.ts

```diff
@@ -28,8 +28,12 @@ export const useDetour = (input: UseDetourInput) => {
   // Record snapshots when changed
   useEffect(() => {
     const snapshotSubscription = actorRef.subscribe(() => {
-      const serializedSnapshot = JSON.stringify(actorRef.getPersistedSnapshot())
+      const persistedSnapshot = actorRef.getPersistedSnapshot()
+      const serializedSnapshot = JSON.stringify(persistedSnapshot)
       localStorage.setItem("snapshot", serializedSnapshot)
+      if (persistedSnapshot.context.uuid) {
```

Comment:
suggestion: I think what this conditional is getting at is that we don't want to save the detour unless we're in certain states. This code is currently using `uuid` as a sentinel value/proxy for "should we save the detour?"

What if instead, we use a tag, maybe called `no-save`, and do `snapshot.hasTag("no-save")` as the guard for if we call `putDetourUpdate`? 

I'll admit I played around with that idea when I built the localStorage stuff, so I'd be happy to pair on it if that'd be helpful.

Reply:
Ohhh I forgot to apply this. Given my other updates, would you still recommend using a tag?

Reply:
Gosh did I try to make this work. But since we moved the `putDetourUpdate` to the state machine, there seems to be NO way of incorporating `hasTag` with a guard, and I'm so so frustrated about it! My "solution" ends up being pretty janky to get the same effect 😒

Reply:
dang that sucks, there's even [unofficial suggestions that there _should_ be a `hasTag`](https://github.com/statelyai/xstate/discussions/4347#discussioncomment-7404932) guard 😭

Reply:
I knooooowwwww I saw that toooo ugh